### PR TITLE
RIA-12603 Move to java 12 and RA 3.1

### PR DESCRIPTION
**Pull request by DmitryNAleshin**
https://smartbear.atlassian.net/browse/RIA-12603

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -48,20 +53,26 @@
...
                         <groupId>org.codehaus.groovy</groupId>
                         <artifactId>groovy-eclipse-compiler</artifactId>
-                        <version>3.4.0-sb-fixed</version>
+                        <version>3.5.0-sb-fixed</version>
                     </dependency>
                     <dependency>
                         <groupId>org.codehaus.groovy</groupId>
                         <artifactId>groovy-eclipse-batch</artifactId>
-                        <version>2.4.3-01</version>
+                        <version>2.4.17-01</version>
```
Comment (antone-sb) [inline pom.xml]: The current version we use is 2.4.17-02.

Comment (DmitryNAleshin) [reply pom.xml]: I've picked this version from RA pom

Comment (antone-sb) [reply pom.xml]: https://github.com/SmartBear/axm/blob/5c37d4d1eb794eb33c0966629c361cfc82d51953/pom.xml#L46

File: pom.xml [hunk 2]
```diff
@@ -37,6 +37,11 @@
             <artifactId>org.eclipse.paho.client.mqttv3</artifactId>
             <version>1.2.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bctsp-jdk15</artifactId>
+            <version>1.46</version>
```
Comment (antone-sb) [inline pom.xml]: @Alexander-Paeshin is it the correct version?

Comment (DmitryNAleshin) [reply pom.xml]: I've picked it from ra pom as well.

Comment (Alexander-Paeshin) [reply pom.xml]: No, it is not correct version. The library should be bctsp-jdk15on, version 1.64

**Pull request merged**